### PR TITLE
Deep linking

### PR DESCRIPTION
**Pull request by kdaveed**
The pull request starts to solve issue #41. Not all of the addressed problems have been solved yet. The following images illustrate the current functionality.

- The index page redirects us to the first resource's page:

screen shot 2017-08-27 at 22 01 55

- The application can handle the /{resourceType}/{id} url pattern, and shows the edit window of the resource:

screen shot 2017-08-27 at 22 02 12

- If the resource id does not exist we got an error message and the table is shown:

screen shot 2017-08-27 at 22 06 46

- The link 'Show' in each resource redirects us to profile page of an individual, which looks as follows:

screen shot 2017-08-27 at 22 02 24

- The entries listed under the relations work as link. For example link to the profile page of a category:
 
screen shot 2017-08-27 at 22 02 36

### CODE REVIEW

File: src/components/cs-index/cs-index.js.coffee [hunk 1]
```diff
@@ -37,38 +40,62 @@ app.directive "csIndex", ['ResourceService', 'csDataStore', 'csRestApi', 'csSett
           resource.$index({include: '*'}).then(
             # successCallback
             (items) ->
+              resource.loaded = true
               $scope.collection = items
+              resource.data = items
```
Comment (rozner) [inline src/components/cs-index/cs-index.js.coffee]: Resource is a global object. No state should be added to it.

File: src/components/cs-profile/cs-profile.js.coffee [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          field = $scope.resources[type].descriptor.fields[0].attribute
+          return item.attributes[field]
+        
+        $scope.go = (id, type) ->
+          csRoute.go("show", {resource : type, id : id})
+
+        getArray = (relationships) ->
+          if relationships == undefined
+            return []
+          else if relationships.data.constructor == Array
```
Comment (rozner) [inline src/components/cs-profile/cs-profile.js.coffee]: this is a valid approach, but maybe we should consider using the `get_associations` methods of the resource and not parse the data directly.

File: src/cs-descriptor-service/cs-descriptor-service.js.coffee [hunk 3]
```diff
@@ -42,6 +42,8 @@ app.service 'csDescriptorService', [ '$q', '$http', 'ResourceService', 'csResour
       @endpoint = data.endpoint
       @base_url = data.base_url
       @descriptor = _.omit data, ['endpoint', 'base_url']
+      @loaded = false
+      @data = null
```
Comment (rozner) [inline src/cs-descriptor-service/cs-descriptor-service.js.coffee]: what is the purpose of this property?

File: src/cs-route-provider/cs-route-provider.js.coffee [hunk 4]
```diff
@@ -0,0 +1,43 @@
+app = angular.module('cloudStorm.routeProvider', [])
+
+app.provider 'csRoute', [ '$stateProvider', ($stateProvider) ->
+   
+  @state 
+   
+  @go = (type, params, options) ->
+    @state.go(type, params, options)
+ 
+  @transitionTo = (params) ->
```
Comment (rozner) [inline src/cs-route-provider/cs-route-provider.js.coffee]: Is this dead code?

File: bin/sample.html [hunk 5]
```diff
@@ -28,9 +29,14 @@
   </head>
 
   <body ng-controller="MainCtrl">
+  
+    <a ui-sref="about({resourceType : 'categories'})" ui-sref-active="active">Categories</a>
```
Comment (rozner) [inline bin/sample.html]: what is `about`?

File: bin/sample.html [hunk 6]
```diff
@@ -9,12 +9,13 @@
...
     <script src="https://code.angularjs.org/1.5.8/angular.js"></script>
-    <script src="https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.8/angular-animate.js"></script>
+   	<script src="https://cdnjs.cloudflare.com/ajax/libs/angular.js/1.5.8/angular-animate.js"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-bootstrap/1.3.3/ui-bootstrap-tpls.min.js"></script>
     <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-select/0.20.0/select.js"></script>
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-select/0.20.0/select.css" />
     <script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>
-
+    <script src="//angular-ui.github.io/ui-router/release/angular-ui-router.js"></script>
+	  <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-ui-router/1.0.0-beta.3/angular-ui-router.js"></script>
```
Comment (rozner) [inline bin/sample.html]: Is this a duplication?

File: src/components/cs-index/cs-index.js.coffee [hunk 7]
```diff
@@ -15,17 +15,20 @@ app.directive "csIndex", ['ResourceService', 'csDataStore', 'csRestApi', 'csSett
 
     # Pre-link: gets called for parent first
     pre: ($scope, element, attrs, controller) ->
-      return
-
+      returns
```
Comment (rozner) [inline src/components/cs-index/cs-index.js.coffee]: syntax error?

**Comment by bence-nagy** [comment]
Closing without merge.
See https://github.com/cloudstorm/cloudstorm/pull/86

**Pull request closed without merging**